cEP-0024.md

```diff
@@ -0,0 +1,226 @@
+# Implement GitMate automations and plugins for coala
+
+| Metadata |                                                       |
+| -------- | ----------------------------------------------------- |
+| cEP      | 24                                                    |
+| Version  | 1.0                                                   |
+| Title    | Implement Gitmate automations and plugins for coala   |
+| Authors  | Vamshi Krishna Bommerla <mailto:bommvams72@gmail.com> |
+| Status   | Proposed                                              |
+| Type     | Process                                               |
+
+## Abstract
+
+This cEP describes the improvements to the coala workflow and the necessary
+automation in [GitMate](https://gitmate.io/) to realise it, as a part of the
+[GSoC project](https://summerofcode.withgoogle.com/projects/#5344316227059712).
+
+## Introduction
+
+[GitMate](https://gitmate.io/) is an automation tool for developers. While it
+works well, there are a number of missing features which would save time of
+coala community, for example automatically close PRs opened on master of forks.
+Many of the plugins in GitMate are general plugins, which are required by every
+organization, but coala needs more plugins which support coala development
+workflow. This project adds plugins to auto-review, auto-reject, auto-label
+PRs, assign, unassign issues, squash commits.
+
+## Changes in coala workflow
+
+### Use increasing difficulty level order to assign issues
+
+Current assign process in coala is to check team membership of a user,
+difficulty level of the issue and assign issue to user if eligible. To
+implement current assign process in GitMate, IGitt should support teams,
+but it is difficult to implement, because there are no teams in GitLab.
+
+Considering teams as groups in GitLab is a possible solution for IGitt
+implementation, but requires extra sync tools to sync GitHub team and GitLab
+group memberships, which nobody has built yet.
+
+Team membership on GitHub is not publicly accessible information, so any system
+which wants to implement the current 'assign' system needs to have a master
+org key, which isn't good.
```

Comment:
A lot of this new section needs to have the wording improved, as it is heavily based on the phrasing used in our private chat which wasnt very polished as it was 'stream of thought' phrasing.  But it has been nicely re-organised here, and just needs some grammar improvements and some 'tone' improvement.